F con fix

TomMaullin opened the pull request:
The aim of this PR is to fix issue #118 by displaying F contrasts correctly.

pep8speaks commented:
Hello @TomMaullin! Thanks for submitting the PR.

 - In the file [`nidmfsl/fsl_exporter/fsl_exporter.py`](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py), following are the PEP8 issues :

> [Line 29:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L29): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 30:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L30): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 31:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L31): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 32:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L32): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 33:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L33): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 34:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L34): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 87:9](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L87): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare except'
> [Line 119:80](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L119): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 120:80](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L120): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 139:80](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L139): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 145:9](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L145): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare except'
> [Line 241:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L241): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 245:47](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L245): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 247:47](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L247): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 248:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L248): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 255:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L255): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 258:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L258): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 260:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L260): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 262:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L262): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 265:17](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L265): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 268:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L268): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 270:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L270): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 276:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L276): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 279:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L279): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 287:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L287): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 291:65](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L291): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 292:37](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L292): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 294:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L294): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 297:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L297): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 299:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L299): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 303:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L303): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 307:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L307): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 310:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L310): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 311:75](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L311): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 318:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L318): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 320:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L320): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 323:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L323): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 326:58](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L326): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 327:50](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L327): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 328:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L328): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 332:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L332): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 335:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L335): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 338:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L338): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 358:80](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L358): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 369:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L369): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 371:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L371): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 397:57](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L397): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 425:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L425): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 472:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L472): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 495:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L495): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 501:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L501): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 513:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L513): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 515:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L515): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 520:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L520): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 525:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L525): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 527:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L527): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 531:64](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L531): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 532:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L532): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 537:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L537): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 545:80](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L545): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (113 > 79 characters)
> [Line 545:81](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L545): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 545:82](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L545): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 578:1](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L578): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 579:17](https://github.com/incf-nidash/nidmresults-fsl/blob/5a686f64ab1561cfcc30ea720e4f1969abd26a69/nidmfsl/fsl_exporter/fsl_exporter.py#L579): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment

TomMaullin commented:
Hi @cmaumet ,

This PR is now ready for review. I shall upload an example dataset to a PR on the NIDMResults-Examples repository this afternoon in order to show it now functions correctly and comment on this PR again when this is done. 

The Travis tests pass but the PEP8 bot still has some complaints. These are not with code I have touched and I am wary of changing anything in these lines as I don't have full knowledge of what is going on in the rest of the code.

TomMaullin commented:
Hi @cmaumet ,

Just to confirm, an example dataset for testing this PR is now available in the below PR:

https://github.com/incf-nidash/nidmresults-examples/pull/101

cmaumet commented:
Hi @TomMaullin! Thanks a lot for this update. Can you provide a short description (and possibly an itemized list) of the changes included in this PR? Thanks!

TomMaullin commented:
Hi @cmaumet , yes sure!

The issue this PR aims to address is issue #118. The cause of issue #118 is probably easiest to explain by first explaining what the master branch currently does for recording contrast names:

- First, it finds a contrast map and records it's number (i.e. if it finds a excursion set named `ExcursionSet_T002` it records the variable `con_num` as 2).
- Then, it looks through the `design.fsf` for the contrast's name, using the variable `con_num` to identify it (see [line 249](https://github.com/incf-nidash/nidmresults-fsl/blob/master/nidmfsl/fsl_exporter/fsl_exporter.py#L249) of `fsl_exporter.py`) and records it as `self.contrast_names_by_num[con_num]`.
- Next it looks through the `design.fsf` for the contrast's weight matrix, again using `con_num` to identify the contrast.

This works very well for T contrasts. The problem with this is that it doesn't account for F contrasts. F contrasts are not recorded in the same way in the `design.fsf`. Say, there are three excursion sets 'ExcursionSet_T001`, 'ExcursionSet_T002` and `ExcursionSet_F001`. Let the F test be some combination of the two T tests. 

For T001;
- `con_num` will be assigned to 1.
- The code will identify it's name as whatever the contrast corresponding to `con_num=1` was called, say `T contrast 1`, by searching the `design.fsf`.
- The code will identify whatever the contrast weight matrix corresponding to `con_num=1` was, say `[1 0]`. 
- This information will be saved and exported as a contrast object later on.

For T002;
- `con_num` will be assigned to 2.
- The code will identify it's name as whatever the contrast corresponding to `con_num=2` was called, say `T contrast 2`, by searching the `design.fsf`.
- The code will identify whatever the contrast weight matrix corresponding to `con_num=2` was, say `[0 1]`. 
- This information will be saved and exported as a contrast object later on.

For F001;
- `con_num` will be assigned to 1.
- The code will identify it's name as whatever the contrast corresponding to `con_num=1` was called. This will again be `T contrast 1` as searching the `design.fsf` in the way the code currently does only gives T contrasts.
- The code will identify whatever the contrast weight matrix corresponding to `con_num=1` was. Again this will be `[1 0]`, which is the contrast weight matrix of the T contrast and not the F which we are interested in. 
- This information will be saved and exported as a contrast object later on.

This is exactly what happens for the dataset [`fsl_con_f`](https://github.com/incf-nidash/nidmresults-examples/tree/master/fsl_con_f). You will see that the F contrast is an exact duplicate of the T contrast. The reason this hasn't been picked up on before is that it just so happens that in this dataset, the F contrast actually was just a duplicate of the T contrast! 

The current code does not record F tests correctly. This can be easily verified by running a case with more F tests than T tests. When there is no T contrast with the same `con_num` as the F contrast the code errors and no export is created.

It is also worth noting though that as the F contrast weight matrix was previously recorded, degrees of freedom for F tests could not be calculated also (as the rank of the contrast weight matrix could not be correctly recorded). 

To rectify this I have changed the following:
- Changed how the F contrast names are obtained.
- Constructed the correct contrast weight matrix for F tests using the contrast weight matrices found for the T tests.
- Updated the way clusters and peaks are read in from the cluster/peak table for excursion sets (as the cluster/peak tables have a different number of columns for T contrasts and F contrasts and the previous indexing only worked for T contrasts).
- Ensured the parameter estimate weights are correctly recorded for F contrasts.
- Record the effect degrees of freedom for F tests.

I hope this explains these changes well. Let me know if you have any further questions.

cmaumet commented:
Hi @TomMaullin! Thanks for the detailed explanation. This looks like an important bug fix. Thanks for tracking this down and submitting this update! Can you rerun the travis testing to make sure that they still pass now that we have updated nidmresults-examples to include your multi-F-contrast example?

(Note: in the future, if there is another instance where you want to test against a branch of the nidmresults-examples repository you can update the `NIDM_EX` variable from the `.travis.yml`).

TomMaullin commented:
Hi @cmaumet ,

I have rerun the travis tests and encountered a slight issue. The `fsl_con_f_multiple` dataset was created using FSL 5.0.10 but I hadn't realized until now that the travis tests are running FSL 5.0.8. For this reason, when the exporter tests are run on `fsl_con_f_multiple_130` they fail but when they are run on a computer with FSL 5.0.10 installed they pass.

To fix this I am going to do the following:
- Make a PR which replaces the current `fsl_con_f_multiple_130` on the `NIDMResults-examples` repository with one obtained from running FSL 5.0.8 on the same `design.fsf`.
- Make a PR to your `Upgrade to FSL 5.0.10 #103` `fsl_5.0.10` branch containing the FSL 5.0.10 version of `fsl_con_f_multiple`. 

Is this plan okay by you? Do you have any alternative suggestions?

cmaumet commented:
Hi @TomMaullin! Thanks for looking into this!

> For this reason, when the exporter tests are run on fsl_con_f_multiple_130 they fail but when they are run on a computer with FSL 5.0.10 installed they pass.

Can you clarify what happens when the tests fail? Which error message? 

> Is this plan okay by you? Do you have any alternative suggestions?

Your plan sounds good! We can then deal with integration with FSL 5.0.10 in another PR.

cmaumet commented:
Discussed with @TomMaullin today:

The tests here are failing because the fsl_con_f_multiple example was created with FSL 5.0.10 and the tests are run using FSL 5.0.8 (from neurodebiam). When the cluster command is called (based on the log created with 5.0.10) they fail with FSL 5.0.8.

It's seems fragile to rely on the `cluster` command inside the FSL exporter (using the command extracted from the log) as current version of FSL might be different with the one used to do the analysis.

Our solution is to remova call to `cluster`cf #114. #114 will have to be solved before we go on with this.

TomMaullin commented:
Hi @cmaumet ,

This PR has now been updated to work with the 2.00 update changes you made recently. However, I am struggling to address the issue created by the call to cluster as the `fsl_con_f_multiple` example is no longer in the nidmresults-examples repository (and the tests take too much memory to run on my laptop at the hackathon).

I was under the impression that this test was going to be temporarily removed from the test script (e.g. maybe by downloading `nidmresults-examples` and then deleting the folder `fsl_con_f_multiple` before running the tests in the `.travis.yml`) but it is no longer in the examples repository. I can't quite remember fully... I know at one point we did talk about moving it... does it now live on a different branch? (i.e. should I now point the travis tests somewhere else?). Or has the entire examples repository been updated with 2.0 as well?

cmaumet commented:
@TomMaullin: could you rebase on master and put back `fsl_con_f_multiple` test (i.e. revert https://github.com/incf-nidash/nidmresults-fsl/pull/124/commits/a3a7ade329eef16a5f0ef0d6d6c2e5479e199132) to see if the tests now pass?
